components/eventing-controller/testing/subscriber.go

```diff
@@ -26,20 +36,108 @@ type Subscriber struct {
 
 type SubscriberOption func(*Subscriber)
 
-const (
-	maxNoOfData           = 10
-	maxAttempts           = 10
-	storeEndpoint         = "/store"
-	checkEndpoint         = "/check"
-	internalErrorEndpoint = "/return500"
-	checkRetriesEndpoint  = "/check_retries"
-)
+// NewSubscriber creates a simple test Subscriber with http endpoints to received, store end answer
```

Comment:
(nit)

Not sure why you change the sequence of the functions, I guess it really makes more sense to have the Constructor sooner in the code. My small problem is, that with such changes you hardly can see the code differences, if there were some.